src/main/java/com/salesforce/mirus/HerderStatusMonitor.java

```diff
@@ -104,29 +103,35 @@ private void processConnector(String connectorName) {
                 }
               });
         }
-      } else {
-        herder.connectorInfo(connectorName, this::onConnectInfo);
       }
+    } else {
+
+      herder.connectorInfo(
+          connectorName,
+          (error, connectorInfo) -> {
+            if (error != null) {
+              logger.warn("Failed to retrieve connector info, Error details: {}", error);
+              return;
+            }
+            // Only the worker with the active controller should report connector metrics
+            if (workerId.equals(stateInfo.connector().workerId())) {
+              connectorJmxReport.handleConnector(herder, connectorInfo);
+            }
+            connectorInfo.tasks().forEach(task -> processTask(task, herder.taskStatus(task)));
```

Comment:
Could you not move this into the if clause and keep `processTask` the same?

Reply:
We want `processTask` to be able to run on all workers - this is actually reverting processTask to the older logic from the previous release.